icefoganalytics/travel-authorization
# Update node version

Relates to:
- https://github.com/icefoganalytics/travel-authorization/issues/116
- #209 

## Context
Switching to a newer version of vitest requires a newer version of node

## Implementation
- Updated dev docker container node version
- Updated prod docker container node version
- [Removed version in docker compose](https://docs.docker.com/compose/intro/history/#:~:text=Version%20two%20of,file.)
- Updated tedious and vue/cli becuase of an azure engine warning

## Screenshots
![2024-07-23_12-29](https://github.com/user-attachments/assets/ed5f35e1-19ac-4b29-aa9e-49ab5d67add3)
*Testing after rebuilding*
# Testing Instructions
1. Run `dev down`
2. Create `.env` in top level directory with the following
```
DB_HOST="db"
DB_PORT="5432"
DB_USER="app"
DB_PASS="itsallgood"
DB_NAME="travel_production"
AWS_LOGGING_ENABLED="false"
```
3. Copy `.env` to `api/.env.production`
4. Remove `api/node_modules` and `web/node_modules`
5. Run `docker compose build --no-cache`
6. Run `docker compose up --remove-orphans`
7. Visit app on localhost:3000, check that you can log in.
